src/Analyzers/Core/Analyzers/AbstractCodeQualityDiagnosticAnalyzer.cs

```diff
@@ -46,20 +47,20 @@ public bool OpenFileOnly(OptionSet options)
 
         protected static DiagnosticDescriptor CreateDescriptor(
             string id,
+            EnforceOnBuild enforceOnBuild,
             LocalizableString title,
             LocalizableString messageFormat,
             bool isUnnecessary,
             bool isEnabledByDefault = true,
             bool isConfigurable = true,
-            LocalizableString description = null,
-            params string[] customTags)
+            LocalizableString description = null)
             => new(
                     id, title, messageFormat,
                     DiagnosticCategory.CodeQuality,
                     DiagnosticSeverity.Info,
                     isEnabledByDefault,
                     description,
                     helpLinkUri: DiagnosticHelper.GetHelpLinkForDiagnosticId(id),
-                    customTags: DiagnosticCustomTags.Create(isUnnecessary, isConfigurable, customTags));
+                    customTags: DiagnosticCustomTags.Create(isUnnecessary, isConfigurable, enforceOnBuild));
```

Comment:
Shouldn't that be `enforceOnBuild.ToCustomTag()`?

Reply:
That is done in `DiagnosticCustomTags.Create` method.